exporter/awsemfexporter/metric_translator.go

```diff
@@ -140,48 +140,116 @@ func TranslateOtToCWMetric(rm *pdata.ResourceMetrics, config *Config) ([]*CWMetr
 	if len(namespace) == 0 {
 		namespace = defaultNameSpace
 	}
+	return namespace
+}
 
-	ilms := rm.InstrumentationLibraryMetrics()
-	for j := 0; j < ilms.Len(); j++ {
-		ilm := ilms.At(j)
-		if ilm.IsNil() {
-			continue
-		}
-		if ilm.InstrumentationLibrary().IsNil() {
-			instrumentationLibName = noInstrumentationLibraryName
-		} else {
-			instrumentationLibName = ilm.InstrumentationLibrary().Name()
-		}
+// TranslateOtToGroupedMetric converts OT metrics to GroupedMetric format
+func TranslateOtToGroupedMetric(metric pdata.Metrics, groupedMetricMap map[string]*GroupedMetric) {
+	totalDroppedMetrics := 0
 
-		metrics := ilm.Metrics()
-		for k := 0; k < metrics.Len(); k++ {
-			metric := metrics.At(k)
-			if metric.IsNil() {
-				totalDroppedMetrics++
+	var dps DataPoints
+	rms := metric.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		rm := rms.At(i)
+		namespace := GetNamespaceFromMetric(&rm)
+		ilms := rm.InstrumentationLibraryMetrics()
+		for j := 0; j < ilms.Len(); j++ {
+			ilm := ilms.At(j)
+			if ilm.IsNil() {
 				continue
 			}
```

Comment:
I think you're missing the `Nil` check for `ilm.InstrumentationLibrary` here. Was there a reason you removed the logic for `instrumentationLibraryName`?

Reply:
added nil check